src/main/java/de/rondiplomatico/nds/NDSCoordinate.java

```diff
@@ -96,6 +96,21 @@ public NDSCoordinate(long ndsMortonCoordinates) {
         latitude = lat;
         longitude = lon;
     }
+    
+    /*
+     * Compute midpoint of this and given NDSCoordinate
+     * 
+     * @param p
+     *          the given coordinate
+     */
+    public NDSCoordinate getMidpoint(NDSCoordinate p) {

```

Comment:
Please make this method more general using a fraction in [0,1]. (verification only via "assert" for efficiency)

```suggestion
    public NDSCoordinate fractionBetween(NDSCoordinate p, double fraction) {
```